pyproject.toml

```diff
@@ -59,16 +59,25 @@ onnxoptimizer = "0.3.13"
 # https://onnxruntime.ai/docs/reference/compatibility.html#onnx-opset-support
 onnxruntime = "1.18"
 hummingbird-ml = {version="0.4.11", extras = ["onnx"]}
-scikit-learn = "1.1.3"
-scipy = "1.10.1"
-numpy = "1.24.2"
-protobuf = "^4.25.2"
+scikit-learn = [
+        {version = "1.1.3", python = "<3.9" }, 
+        {version = "1.4.0", python = ">=3.9.0"}        
+        ]
+scipy = [
+        {version = "1.10.1", python = "3.8"},
+        {version = ">=1.10.1", python = ">3.8"}
+        ]
+numpy = [
+        {version = "1.23.5", python = "<3.9" }, 
+        {version = "1.26.4", python = ">=3.9.0"}        
+        ]
+protobuf = "3.20.3"
```

Comment:
I'm not sure why we need these specific versions now.. didn't we need them already when supporting python3.11 ?